commit/gasworker/outcome.go

```diff
@@ -0,0 +1,15 @@
+package gasworker
+
+import (
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+)
+
+func (p *GasWorker) Outcome(
+	outCtx ocr3types.OutcomeContext, observations []GasObservation,
```

Comment:
Feels like this may need to be `AttributedObservation` to provide the most possible info to the sub-plugins, i.e maybe the oracle ID is required in order to log some information

Reply:
The main reason I didn't do it like that is because we'll have n workers decoding it, instead I decode it once in the plugin and send only the required info to the worker.

Reply:
I think it's just abstracting away something that doesn't really need to be abstracted IMO.

Reply:
Doesn't have to be encoded, it can be something like this:

```golang
type AttributedObservation[ObservationType any] struct {
  OracleID uint8
  Observation ObservationType
}
```